bftengine/src/bftengine/ReplicaConfigSerializer.cpp

```diff
@@ -0,0 +1,208 @@
+//Concord
+//
+//Copyright (c) 2019 VMware, Inc. All Rights Reserved.
+//
+//This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+//This product may include a number of sub-components with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the LICENSE
+// file.
+
+#include "ReplicaConfigSerializer.hpp"
+#include "threshsign/IThresholdSigner.h"
+#include "threshsign/IThresholdVerifier.h"
+#include "../../src/bftengine/SysConsts.hpp"
+
+using namespace std;
+using namespace concordSerializable;
+
+namespace bftEngine {
+namespace impl {
+
+uint32_t ReplicaConfigSerializer::maxSize(uint32_t numOfReplicas) {
+  return (sizeof(config_->fVal) + sizeof(config_->cVal) +
+      sizeof(config_->replicaId) +
+      sizeof(config_->numOfClientProxies) +
+      sizeof(config_->statusReportTimerMillisec) +
+      sizeof(config_->concurrencyLevel) +
+      sizeof(config_->autoViewChangeEnabled) +
+      sizeof(config_->viewChangeTimerMillisec) + MaxSizeOfPrivateKey +
+      numOfReplicas * MaxSizeOfPublicKey +
+      IThresholdSigner::maxSize() * 3 +
+      IThresholdVerifier::maxSize() * 3);
+}
+
+void ReplicaConfigSerializer::registerClass() {
+  SerializableObjectsDB::registerObject("ReplicaConfig", SharedPtrToClass(new ReplicaConfigSerializer));
+}
+
+ReplicaConfigSerializer::ReplicaConfigSerializer(ReplicaConfig *config) {
+  config_ = new ReplicaConfig;
+  if (config)
+    *config_ = *config;
+  registerClass();
+}
+
+ReplicaConfigSerializer::ReplicaConfigSerializer() {
+  config_ = new ReplicaConfig;
+}
+
+ReplicaConfigSerializer::~ReplicaConfigSerializer() {
```

Comment:
Please follow the rule of 3 and add copy/copy-assignment methods or mark them as deleted. Alternatively, use a unique_ptr for config_. The latter is preferable as it makes the code smaller.

Reply:
OK.